### PR TITLE
cpu/stm32: Fix stm clock configuration

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
### Contribution description

This splits the stm32 clock tree in Kconfig making it a bit easier to look at and manage.
It exposes the clock speeds allowing enough information to match the make resolution.
It also introduces some `HAVE_` symbols to make it a bit easier to manage multiple CPU_SERIES and/ors.

There is a small fix to the make resolution for the stm32f303*, some boards were not running at the max possible clock speed.

### Testing procedure

Green murdock with `NIGHTLY=1`

Self testing based off #18771 with
```
export TEST_BOARDS=$(FEATURES_REQUIRED=cpu_stm32 make -C tests/pkg_tinyusb_cdc_msc info-boards-supported --no-print-directory)
for board in $TEST_BOARDS; do echo $board && /bin/bash -c "source .murdock; JOBS=8 compile tests/pkg_tinyusb_cdc_msc ${board}:gnu" | grep mismatch; done;
```
as well (since this should fix the USB errors)

### Issues/PRs references

Follow up #18771

### CODE REVIEW

**Comment by MrKevinWeiss** [comment]
Note that I still have to clean up commits...

File: boards/nucleo-wl55jc/Kconfig [hunk 1]
```diff
@@ -22,10 +22,16 @@ config BOARD_NUCLEO_WL55JC
     select HAS_PERIPH_TIMER
     select HAS_PERIPH_UART
 
+    select BOARD_HAS_HSE
```
Comment (Teufelchen1) [inline boards/nucleo-wl55jc/Kconfig:25]: Misses `# Clock configuration` comment as seen on other boards

File: cpu/stm32/dist/clk_conf/clk_conf.c [hunk 2]
```diff
@@ -454,7 +454,8 @@ int main(int argc, char **argv)
     printf("#define CLOCK_CORECLOCK     (%uU)\n", coreclock);
     printf("/* 0: no external high speed crystal available\n"
            " * else: actual crystal frequency [in Hz] */\n"
-           "#define CLOCK_HSE           (%uU)\n", pll_src ? pll_in : 0);
+           "#ifndef CONFIG_CLOCK_HSE
+#define CONFIG_CLOCK_HSE           (%uU)\n", pll_src ? pll_in : 0);
```
Comment (Teufelchen1) [inline cpu/stm32/dist/clk_conf/clk_conf.c:458]: Formatting looks odd?

File: cpu/stm32/stm32_line.mk [hunk 3]
```diff
@@ -56,6 +56,7 @@ ifeq (F,$(STM32_TYPE))
       ifneq (, $(filter $(STM32_ROMSIZE), 6 8))
         CPU_LINE = STM32F$(STM32_MODEL)x8
       else ifneq (,$(filter $(STM32_ROMSIZE), B C))
+        $(info ************  TEST VERSION ************)
```
Comment (Teufelchen1) [inline cpu/stm32/stm32_line.mk:59]: What's its purpose?

File: boards/nucleo-wl55jc/Kconfig (hunk 1 above)
Comment (MrKevinWeiss) [reply boards/nucleo-wl55jc/Kconfig:25]: Removed as it is defined in the `source "$(RIOTBOARD)/common/nucleo64/Kconfig"`, maybe that gets more explicit later...

**Comment by MrKevinWeiss** [comment]
Commits have been cleaned up, probably supersedes #18771, all in one PR splitting is a bit of a pain.

**Comment by MrKevinWeiss** [comment]
Note: I should not make the `CLOCK_HSE` value selectable and remove the `CLOCK_HSE_*MHZ` options.

**Comment by MrKevinWeiss** [comment]
If the nightly murdock results are good I would remove the REMOVEME commit and say it is ready...

**Comment by MrKevinWeiss** [comment]
Anyone else willing to review?  This is needed to fix nightlies? if @aabadie doesn't have any strong opinions then maybe we can move with this?

**Comment by MrKevinWeiss** [comment]
So this indicates that it does fix the intended failures on the nightlies.  Can I squash and someone take a look?

**Comment by leandrolanzieri** [comment]
Go ahead

**Comment by MrKevinWeiss** [comment]
All passing...

**Comment by MrKevinWeiss** [comment]
Now that #18787 is merged, this is the only remaining fix needed for nightlies.

**Review by maribu — APPROVED** [review]
ACK. Changes look good to me.

Testing doesn't scale to all of the STM32 boards, so I kind of fear this will break some board. But just letting this sitting here for all eternity while nightlies remain failing is not really an option. So let's merge this and fix the fallout later when the bug reports come in.

**Comment by MrKevinWeiss** [comment]
Thanks, this is mostly just matching the kconfig to the header dependency resolution, so low risk from the "production" side.  I noticed a few things that were corrected though.

I also tested against all stm boards with the compile_like_murdock.py script that is PRed as well.  Not all apps but the ones that would be affected by USB and non-USB clock contraints.

**Pull request merged**